lms/views/api/vitalsource.py

```diff
@@ -0,0 +1,27 @@
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.view import view_config, view_defaults
+
+from lms.security import Permissions
+from lms.views.api.vitalsource_sample_data import book_list, toc_data
+
+
+@view_defaults(renderer="json", permission=Permissions.API)
+class VitalSourceAPIViews:
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(route_name="vitalsource_api.books.list")
+    def list_books(self):  # pylint:disable=no-self-use
+        # In future this should, somehow, get the list of books associated with
+        # the current course.
+        return book_list
+
+    @view_config(route_name="vitalsource_api.books.toc")
+    def table_of_contents(self):
+        book_id = self.request.matchdict["book_id"]
+
+        # In future this will fetch the data from VitalSource using the
+        # https://api/vitalsource.com/v4/products/{BOOK_ID}/toc endpoint.
+        if book_id not in toc_data:
+            raise HTTPNotFound("Table of contents not available for book")
```

Comment:
How does the client handle this 404 error? What error dialog is displayed? In general I think `404`'s aren't expected from the API and `400` with a JSON body should be used. See https://github.com/hypothesis/lms/blob/b08d57b2d76d7a76e88afab560e1d848242357c5/lms/views/api/exceptions.py#L25-L91